stdlib/src/builtin/int.mojo

```diff
@@ -1212,6 +1213,64 @@ struct Int(
 
         writer.write(self)
 
+    @staticmethod
+    fn from_bytes[
+        type: DType, big_endian: Bool = False
+    ](bytes: Span[Byte]) raises -> Self:
+        """Converts a byte array to an integer.
+
+        Args:
+            bytes: The byte array to convert.
+
+        Parameters:
+            type: The type of the integer.
+            big_endian: Whether the byte array is big-endian.
+
+        Returns:
+            The integer value.
+        """
+        if type.sizeof() != len(bytes):
+            raise Error("Byte array size does not match the integer size.")
+        var ptr: UnsafePointer[Byte] = UnsafePointer.address_of(bytes[0])
+        var type_ptr: UnsafePointer[Scalar[type]] = ptr.bitcast[Scalar[type]]()
+        var value = type_ptr[]
+
+        @parameter
+        if is_big_endian() and not big_endian:
+            value = byte_swap(value)
+        elif not is_big_endian() and big_endian:
+            value = byte_swap(value)
+        return int(value)
+
+    fn as_bytes[type: DType, big_endian: Bool = False](self) -> List[Byte]:
+        """Convert the integer to a byte array.
+
+        Parameters:
+            type: The type of the integer.
+            big_endian: Whether the byte array should be big-endian.
+
+        Returns:
+            The byte array.
+        """
+        alias type_len = type.sizeof()
+        var value = Scalar[type](self)
+
+        @parameter
+        if is_big_endian() and not big_endian:
+            value = byte_swap(value)
+        elif not is_big_endian() and big_endian:
+            value = byte_swap(value)
+
+        var ptr: UnsafePointer[Scalar[type]] = UnsafePointer.address_of(value)
+        var byte_ptr: UnsafePointer[Byte] = ptr.bitcast[Byte]()
+        var list = List[Byte](capacity=type_len)
+
+        # TODO: Maybe this can be a List.extend(ptr, count) method
+        memcpy(list.unsafe_ptr(), byte_ptr, type_len)
+        list.size = type_len
+
```

Comment:
I don't think we should add more `UnsafePointer` APIs than absolutely necessary
```suggestion
        var ptr =  UnsafePointer.address_of(value)
        var list = List[Byte](capacity=type_len)

        memcpy(list.unsafe_ptr(), ptr.bitcast[Byte](), type_len)
        list.size = type_len
```

Reply:
Agreed. Changed here: https://github.com/msaelices/mojo/commit/30027dffadc04f55616bbc49f4c111ff3f24d50e